src/MySqlConnector/Core/SchemaProvider.cs

```diff
@@ -58,6 +60,54 @@ public DataTable GetSchema(string collectionName)
 			return dataTable;
 		}
 
+		private void FillDataSourceInformation(DataTable dataTable)
+		{
+			dataTable.Columns.AddRange(new [] {
+				new DataColumn("CompositeIdentifierSeparatorPattern", typeof(string)),
+				new DataColumn("DataSourceProductName", typeof(string)),
+				new DataColumn("DataSourceProductVersion", typeof(string)),
+				new DataColumn("DataSourceProductVersionNormalized", typeof(string)),
+				new DataColumn("GroupByBehavior", typeof(GroupByBehavior)),
+				new DataColumn("IdentifierPattern", typeof(string)),
+				new DataColumn("IdentifierCase", typeof(IdentifierCase)),
+				new DataColumn("OrderByColumnsInSelect", typeof(bool)),
+				new DataColumn("ParameterMarkerFormat", typeof(string)),
+				new DataColumn("ParameterMarkerPattern", typeof(string)),
+				new DataColumn("ParameterNameMaxLength", typeof(int)),
+				new DataColumn("QuotedIdentifierPattern", typeof(string)),
+				new DataColumn("QuotedIdentifierCase", typeof(IdentifierCase)),
+				new DataColumn("ParameterNamePattern", typeof(string)),
+				new DataColumn("StatementSeparatorPattern", typeof(string)),
+				new DataColumn("StringLiteralPattern", typeof(string)),
+				new DataColumn("SupportedJoinOperators", typeof(SupportedJoinOperators))
+			});
+
+			var row = dataTable.NewRow();
+			row["CompositeIdentifierSeparatorPattern"] = @"\.";
+			row["DataSourceProductName"] = "MysqlConnector";
+			row["DataSourceProductVersion"] = m_connection.ServerVersion;
+			row["DataSourceProductVersionNormalized"] = m_connection.Session.ServerVersion.Version.ToString();
```

Comment:
> A normalized version for the data source, such that it can be compared with `String.Compare()`.

There would need to be some padding here (e.g., `nn.nn.nnnn`) to make sure 10 doesn't compare between 1 and 2.